### DISCUSSION TITLE
Is this project ever going to be fully open source?

### DISCUSSION THREAD

**Discussion by hnhx** [NVIDIA/open-gpu-kernel-modules#315] [Q&A] [answered]
I wonder if the rest of the components of the NVIDIA driver are going to be fully open source in the future, or is it going to remain like this and only the kernel modules are going to be open source and nothing else.

**Comment by mtijanic** [comment, marked as answer]
Hi there! There are currently no plans to open source the usermode drivers.

**Reply by mtijanic** [reply]
> But, it could be swapped out with Mesa, right?

Yes; it would require changes to Mesa and likely these kernel drivers as well, but I believe there is nothing preventing it from happening, someone just needs to do all the work. I am not aware of any plans to do this at NVIDIA.

**Reply by throwaway1037** [reply]
Approximately how many LOC is the proprietary userspace components?
I doubt reimplementing them with Mesa is going to be an easy or short task.

**Reply by hnhx** [reply]
Thanks for the answer. Well seems like Intel still remains the only choice for those who care about software freedom since both AMD and NVIDIA need closed source firmware / software to work properly. I guess NVIDIA is still better than AMD since you can use an NVIDIA card with nouveau but almost all AMD cards require closed source firmware to work at all with Xorg or Wayland.

**Reply by throwaway1037** [reply]
Nouveau needs the `linux-firmware` blobs and this driver has inlined proprietary blobs: https://github.com/NVIDIA/open-gpu-kernel-modules/discussions/291

**Reply by YusufKhan-gamedev** [reply]
@hnhx: linux-libre allows AMD gpus to use Wayland/X/Mir if you need it.

**Reply by hnhx** [reply]
@YusufKhan-gamedev Most AMD GPUs don't work with Xorg without non free firmware.

@throwaway1037 My NVIDIA RTX 3050 worked fine with nouveau, linux-libre and no nonfree firmware, so thats wrong. rms prefers NVIDIA over AMD since nouveau works without non free firmware, unlike amdgpu.

**Reply by throwaway1037** [reply]
@hnhx I doubt that for a couple of reasons:
1. Whichever OS you are using probably ships proprietary firmware and software.
If you want to confirm it works in freedom, try installing [Parabola](https://parabola.nu), only install software from the Parabola repos, and report back if it still works.
2. Do you have a source for the RMS comment? It's widely agreed-upon that AMD cards are better in freedom than NVIDIA ones, as both require proprietary firmware, except that the AMD ones have libre, mainlined kernel code, unlike NVIDIA which only has this partially free, non-mainlined kernel module.

**Reply by throwaway1037** [reply]
@hnhx also, what do you mean specifically by "worked"? Nouveau provides significantly less perfomace than the proprietary drivers and firmware. Sure, it might boot with Nouveau, but I doubt the perfomance is anything to brag about.

**Reply by jorenham** [reply]
Why?
